packages/strapi-generate-api/templates/mongoose/service.template

```diff
@@ -27,7 +27,7 @@ module.exports = {
     // Generate stages.
     const populateStage = hook.load().generateLookupStage(<%= globalID %>, { whitelistedPopulate: populate }); // Nested-Population
     const matchStage = hook.load().generateMatchStage(<%= globalID %>, filters); // Nested relation filter
-    const aggregateStages = mergeStages(populateStage, matchStage);
+    const aggregateStages = mergeStages(filters.populate || populateStage, matchStage);
```

Comment:
If I'm not mistaken, if you want to populate a given list of relations ['relation1', 'relation2'] you need to pass them to `generateLookupStage` as follow:
```JS
const populateStage = hook.load().generateLookupStage(<%= globalID %>, { whitelistedPopulate: filters.populate || populate });
```

Reply:
Good catch!